### ISSUE TITLE
TeXmacs (option/plugin)

### ISSUE THREAD

**Issue by nilqed** [agraef/pure-lang#36]
The '--texmacs' option does not work, and consequently the plugin doesn't either. Moreover, 
the TeXmacs *.deb packages (https://texmacs.org/tmweb/download/linux-packages.en.html#ubuntu) now install to '/usr/local/share/TeXmacs/', whereas the plugin goes to 
'/usr/share/texmacs/TeXmacs/'. When asked, I recommended to compile/install both, Pure as well as TeXmacs, by hand ;)

kfp@sirius:~$ apt show pure
Package: pure
Version: 0.68-2
Priority: extra
Section: devel
Maintainer: Albert Graef <aggraef@gmail.com>
Installed-Size: 568 kB
Depends: libpure8 (= 0.68-2), libc6 (>= 2.33), libgcc-s1 (>= 3.0), libllvm3.5v5, libreadline8 (>= 6.0), libstdc++6 (>= 5.2)
Suggests: pure-dev, pure-elisp, pure-docs
Homepage: https://agraef.github.io/pure-lang
Download-Size: 162 kB
APT-Manual-Installed: yes
APT-Sources: http://download.opensuse.org/repositories/home:aggraef:pure/xUbuntu_21.04  Packages

kfp@sirius:~$ apt show pure-texmacs
Package: pure-texmacs
Version: 0.68-2
Priority: extra
Section: devel
Source: pure
Maintainer: Albert Graef <aggraef@gmail.com>
Installed-Size: 1’274 kB
Depends: pure (>= 0.68-2)
Homepage: https://agraef.github.io/pure-lang
Download-Size: 82.3 kB
APT-Manual-Installed: yes
APT-Sources: http://download.opensuse.org/repositories/home:aggraef:pure/xUbuntu_21.04  Packages

**Comment by agraef** [comment]
Hi Kurt, sorry for responding so late. The sad truth is that I've not had the time to look after the Pure OBS packages for quite some time now, as I've been busy maintaining other projects I need for my courses. So they might well be broken. Anyway, thanks for the report, I'll have a look asap, but probably not before the end of the semester, i.e., end of July.

**Comment by agraef** [comment]
Finally got the OBS packages up to snuff again. Looking into the texmacs issue now.

**Comment by agraef** [comment]
Hmm, I guess I see what you mean. Running with the Ubuntu 21.10 deb on Ubuntu 22.04, with just the minimal pure session from https://agraef.github.io/pure-docs/install.html#texmacs-plugin, this what I get:

![image](https://user-images.githubusercontent.com/2853977/183266870-ea69fdba-e5f8-4253-b952-4606d91992de.png)

Is this what you get, too?

The Scheme session seems to work fine, so it seems reasonable to assume that the issue is with `pure -i --texmacs`. But all this stuff is working perfectly (with the full Pure session) on Arch. The version I'm running there is 2.1.2, though, whereas the version from https://www.texmacs.org/tmweb/download/linux-packages.en.html#ubuntu is still 2.1.1.

So I guess that the first thing to try is to build TeXmacs 2.1.2 myself from source on Ubuntu, and see whether the problem persists.

**Comment by nilqed** [comment]
The deb version apparently doesn't send the texmacs ESC codes ... 
![ubu](https://user-images.githubusercontent.com/1512802/183268557-2f18b06c-1009-4b56-9b1c-9e15ba843efa.png)

whereas one expects `verbatim:prompt#>`?
![ming](https://user-images.githubusercontent.com/1512802/183268558-d6cfabaa-2ca3-496c-83c6-3c041091802c.png)

I reported this on behalf of some colleagues, and when compiled from scratch, it worked (as I was told), thus low priority ;)

**Comment by agraef** [comment]
Yep, it seems that TeXmacs 2.1.2 (+ guile 1.8.8) works just fine when built from source:

![image](https://user-images.githubusercontent.com/2853977/183269121-407630fb-7c1a-46e2-83f9-335401ff0f9b.png)

**Comment by agraef** [comment]
I'll probably do a proper OBS build of guile 1.8.8 and TeXmacs 2.1.2 as soon as I find the time, but for the time being, here's the build process that I used, so that you can get up and running quickly.

guile (from ftp://ftp.gnu.org/gnu/guile/guile-1.8.8.tar.gz):

~~~
sudo apt install libgmp-dev libltdl-dev
wget https://raw.githubusercontent.com/slowphil/mingw-w64-guile1.8/master/guile-1.8.8-configure.patch
patch -p1 -i guile-1.8.8-configure.patch
./configure --disable-static --disable-error-on-warning
make
sudo make install
sudo ldconfig -v
~~~

TeXmacs (from https://www.texmacs.org/Download/ftp/tmftp/source/TeXmacs-2.1.2-src.tar.gz):

~~~
sudo apt install qtbase5-dev libqt5svg5-dev libfreetype-dev libsqlite-dev libjpeg-dev
mkdir build && cd build
cmake ..
make
sudo make install
~~~

Then go to pure-lang/pure and do:

~~~
sudo make install-tm
~~~

This will find TeXmacs in /usr/local. Or you can install into ~/.TeXmacs:

~~~
make install-tm tmdir=~/.TeXmacs
~~~

**Comment by agraef** [comment]
I've closed this for now, since the problem really lies upstream, and needs to be addressed upstream by providing a proper package. ;-) Don't get me wrong, I love TeXmacs, but the debs on the TeXmacs website aren't really up to snuff, packaged software should go into /usr, never into /usr/local where it interferes with local builds. And then they're distributing old and broken debs for an Ubuntu release that's already EOL. Looks like they should be looking for a Linux package maintainer. :)

**Comment by agraef** [comment]
Reopening because the issue provides useful instructions on how to work around the problem by installing TeXmacs from source. Also it can serve as a reminder that we should address this in the future if the situation with 3rd party packages doesn't improve, by providing suitable packages on the OBS.

**Comment by nilqed** [comment]
As you please; as far as I'm concerned, it is settled.
Many thanks Albert!
